### PR TITLE
[feat] init workflow ci/cd

### PR DESCRIPTION
**Pull request by lucas-debize**

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      working-directory: ./backend
+      run: npm install
+
+    - name: Build
+      working-directory: ./backend
+      run: npm run build
+ 
+    - name: Run Backend Tests
+      working-directory: ./backend
+      run: npm test
```
Comment (alanschnegg) [inline .github/workflows/ci-cd.yml]: npm run test

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: 21
```
Comment (alanschnegg) [inline .github/workflows/ci-cd.yml]: Wrong version

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
```
Comment (alanschnegg) [inline .github/workflows/ci-cd.yml]: Wrong version

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      working-directory: ./frontend
+      run: npm install
+
+    - name: Display Node.js version
+      working-directory: ./frontend
+      run: node -v
+
+    - name: Install Next.js
+      working-directory: ./frontend
+      run: npm install next
```
Comment (alanschnegg) [inline .github/workflows/ci-cd.yml]: Wrong order

**Review by alanschnegg — CHANGES_REQUESTED** [review]

File: backend/package.json [hunk 5]
```diff
@@ -7,7 +7,8 @@
   "scripts": {
     "start": "tsc && node dist/app.js",
     "build": "tsc",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "lint": "exit 0",
```
Comment (alanschnegg) [inline backend/package.json]: Wait for the linter PR and add it to this one

File: frontend/package.json [hunk 6]
```diff
@@ -6,7 +6,8 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "next lint"
+    "lint": "exit 0",
+    "test": "exit 0"
```
Comment (alanschnegg) [inline frontend/package.json]: same

**Review by alanschnegg — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,70 @@
+name: Workflow CI/CD
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - develop
```
Comment (thmarinho) [inline .github/workflows/test.yml:9]: Test with this :
```yml
on:
  push:
  branches:
```

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      with:
+        node-version: 20
+
+    - name: Install Frontend Dependencies
+      working-directory: ./frontend
+      run: npm install
+
+    - name: Install Next.js
+      working-directory: ./frontend
+      run: npm install next
```
Comment (thmarinho) [inline .github/workflows/test.yml]: Why do you need this ?

File: .github/workflows/test.yml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      working-directory: ./frontend
+      run: npm run build
+
+    - name: Run Frontend Tests
+      working-directory: ./frontend
+      run: npm run test
+
+    - name: Run Frontend Linter
+      working-directory: ./frontend
+      run: npm run lint
```
Comment (thmarinho) [inline .github/workflows/test.yml]: Run the linter before building, this a more common error

**Review by alanschnegg — APPROVED** [review]

File: .github/workflows/test.yml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+      working-directory: ./backend
+      run: npm run build
+ 
+    - name: Run Backend Tests
+      working-directory: ./backend
+      run: npm run test
+
+    - name: Run Backend Linter
+      working-directory: ./backend
+      run: npm run lint
```
Comment (thmarinho) [inline .github/workflows/test.yml:54]: Same thing for here

File: .github/workflows/test.yml [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+      working-directory: ./frontend
+      run: npm install next
+
+    - name: Build Frontend
+      working-directory: ./frontend
+      run: npm run build
+
+    - name: Run Frontend Tests
+      working-directory: ./frontend
+      run: npm run test
```
Comment (thmarinho) [inline .github/workflows/test.yml]: Do'nt run tests until frontend has some

**Review by thmarinho — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 7 above)
Comment (lucas-debize) [reply .github/workflows/test.yml:9]: not working without " -'*' "

File: .github/workflows/test.yml [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+      working-directory: ./backend
+      run: npm run lint
+
+    - name: Build
+      working-directory: ./backend
+      run: npm run build
+ 
+    - name: Run Backend Tests
+      working-directory: ./backend
+      run: npm run test
```
Comment (thmarinho) [inline .github/workflows/test.yml]: Delete this, backend's tests hasn't been merged

File: backend/package.json [hunk 13]
```diff
@@ -7,7 +7,7 @@
   "scripts": {
     "start": "tsc && node dist/app.js",
     "build": "tsc",
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "echo \"Error: no test specified\" && exit 0",
```
Comment (thmarinho) [inline backend/package.json]: Restore your changes on this line

**Review by thmarinho — CHANGES_REQUESTED** [review]

File: frontend/pages/component/utils.tsx [hunk 14]
```diff
@@ -1,3 +1,3 @@
-export function classNames(...classes: string[]): string {
+export default function classNames(...classes: string[]): string {
     return classes.filter(Boolean).join(' ');
 }
```
Comment (thmarinho) [inline frontend/pages/component/utils.tsx]: You should remove the `default`  or rename the file, `utils.tsx` -> `classNames.tsx` for instance

**Review by thmarinho — CHANGES_REQUESTED** [review]

**Review by thmarinho — APPROVED** [review]

**Pull request merged**